compiler/transforms/stream_snaxify.py

```diff
@@ -130,12 +130,6 @@ def generate_one_list(n: int, i: int):
 
             spat_dim = streamer_config.data.spatial_dim()
 
-            # extremely dirty fix:
-            # FIXME: this only works because gemm is the only one with
-            # two spat_dims. This must be fixed with some "virtual" spatial dim
-            if spat_dim == 2:
-                spat_dim = 3
-
```

Comment:
Solving this is the main takeaway for this PR. If a spatial dimension is irrelevant to a certain operand, the streamer does not exist and should not be programmed. With the streamer flags this is now properly handled.